### ISSUE TITLE
FitBounds changes y and height when called on osd with display:none parent

### ISSUE THREAD

**Issue by radpet** [openseadragon/openseadragon#957]
If am calling viewport.fitBounds(rect) when the canvas is in display:none parent there is mismatch of the rect I pass and I get after I set the canvas'  to display:block. This happens under all browsers. Any ideas how to fix it?

**Comment by avandecreme** [comment]
> I think it is problem with the aspectRatio and will investigate it further

You are correct. With `display: none` the  width and height of the viewport are 0.
Your best bet is probably to call `viewport.fitBounds(rect)` immediately after removing `display: none`.

**Comment by radpet** [comment]
Yea I was thinking about it, however this is unacceptable solution for now. This problem will stay because of how browsers threat display:none , won't it?...

**Comment by avandecreme** [comment]
Probably yes, except if someone has a good idea to workaround that...

**Comment by iangilman** [comment]
I think if you put `autoResize: false` in your options you can avoid this (assuming your viewer is visible to begin with). At least that might be an avenue worth investigating.
